public/pages/DetectorResults/utils/advancedStatisticalAnalysis.ts

```diff
@@ -0,0 +1,940 @@
+    });
+    
+    // Simplified p-value calculation
+    return Math.exp(-chiSquare / 2);
+  }
+  
+  private performMannWhitneyTest(values1: number[], values2: number[]): number {
+    if (values1.length < 2 || values2.length < 2) return 1;
+    
+    const allValues = [...values1, ...values2];
+    const ranks = this.calculateRanks(allValues);
+    
+    const ranks1 = ranks.slice(0, values1.length);
+    const ranks2 = ranks.slice(values1.length);
+    
+    const sumRanks1 = ranks1.reduce((sum, rank) => sum + rank, 0);
+    const sumRanks2 = ranks2.reduce((sum, rank) => sum + rank, 0);
+    
+    const u1 = sumRanks1 - (values1.length * (values1.length + 1)) / 2;
+    const u2 = sumRanks2 - (values2.length * (values2.length + 1)) / 2;
+    
+    const u = Math.min(u1, u2);
+    const mean = (values1.length * values2.length) / 2;
+    const variance = (values1.length * values2.length * (values1.length + values2.length + 1)) / 12;
+    
+    const z = Math.abs(u - mean) / Math.sqrt(variance);
+    
+    return this.calculateZPValue(z);
+  }
+  
+  private calculateVariance(values: number[]): number {
+    if (values.length < 2) return 0;
+    
+    const mean = values.reduce((sum, val) => sum + val, 0) / values.length;
+    const sumSquaredDiffs = values.reduce((sum, val) => sum + Math.pow(val - mean, 2), 0);
+    
+    return sumSquaredDiffs / (values.length - 1);
+  }
+  
+  private calculateAutocorrelation(values: number[]): number[] {
+    const n = values.length;
+    const autocorr: number[] = [];
+    
+    for (let lag = 0; lag < Math.min(n / 2, 20); lag++) {
+      let numerator = 0;
+      let denominator = 0;
+      
+      for (let i = 0; i < n - lag; i++) {
+        numerator += values[i] * values[i + lag];
+        denominator += values[i] * values[i];
+      }
+      
+      autocorr.push(denominator === 0 ? 0 : numerator / denominator);
+    }
+    
+    return autocorr;
+  }
+  
+  private groupByHour(values: number[], timestamps: number[]): number[] {
+    const hourlyValues: number[] = new Array(24).fill(0);
+    const hourlyCounts: number[] = new Array(24).fill(0);
+    
+    values.forEach((value, index) => {
+      const hour = new Date(timestamps[index]).getHours();
+      hourlyValues[hour] += value;
+      hourlyCounts[hour]++;
+    });
+    
+    return hourlyValues.map((sum, hour) => hourlyCounts[hour] > 0 ? sum / hourlyCounts[hour] : 0);
+  }
+  
+  private calculateTPValue(t: number, df: number): number {
+    // Simplified t-distribution p-value calculation
+    if (df <= 0) return 1;
+    
+    // Approximation for t-distribution
+    const x = t * t / (t * t + df);
+    const p = 1 - this.betaRegularized(x, df / 2, 0.5);
+    
+    return Math.max(0, Math.min(1, p));
+  }
+  
+  private calculateZPValue(z: number): number {
+    // Simplified normal distribution p-value calculation
+    return 2 * (1 - this.normalCDF(Math.abs(z)));
+  }
+  
+  private betaRegularized(x: number, a: number, b: number): number {
```

Comment:
Have you checked whether existing, well-maintained libraries already support these computations? If we implement everything ourselves, we’ll need to fully understand the math and edge cases before taking it to production.